PatchesAndAreasApi/V1/Gateways/PatchesGateway.cs

```diff
@@ -109,7 +109,12 @@ public async Task<PatchesDb> DeleteResponsibilityFromPatch(DeleteResponsibilityF
             if (ifMatch != patch.VersionNumber)
                 throw new VersionNumberConflictException(ifMatch, patch.VersionNumber);
 
-            if (patch.ResponsibleEntities.Count == responsibleEntitiesRequestObject.Count & patch.ResponsibleEntities.AreAllSame(responsibleEntitiesRequestObject))
+            // Check if the list of patches sent exactly matches the list of patches in the database
+            var allNewInOld = patch.ResponsibleEntities.All(responsibleEntitiesRequestObject.Contains);
+            var allOldInNew = responsibleEntitiesRequestObject.All(patch.ResponsibleEntities.Contains);
+            var entitiesMatch = allNewInOld && allOldInNew;
+
+            if (patch.ResponsibleEntities.Count == responsibleEntitiesRequestObject.Count & entitiesMatch)
                 throw new NoChangesException();
```

Comment:
One of the main reasons why I wanted it to throw an exception is because if the request continues it would send a SNS message with no changes which would cause unnecessary costs. Happy for suggestions though

Reply:
Can the SNS broadcasting not just be made optional on the basis of this result? If you do it the way you have defined above, you're using exceptions as part of the business logic. Using exceptions as part of regular business logic is an anti-pattern IMO.

Reply:
removed the check for now and added a comment there so it can be looked at in the future if any such scenarios occurred